Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Performance.hs

```diff
@@ -25,7 +22,7 @@ newtype PerformanceRes = PerformanceRes
 
 getDriverPerformance :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r) => Id SP.Person -> m PerformanceRes
 getDriverPerformance driverId = do
-  _ <- Esq.runInReplica $ QP.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
-  allRefferedCustomers <- QRD.findAllReferredByDriverId driverId
-  let ridesTakenList = filter (.hasTakenValidRide) allRefferedCustomers
-  pure $ PerformanceRes (Results (length allRefferedCustomers) (length ridesTakenList))
+  driverReferral <- SQD.findById driverId
+  case driverReferral of
+    Just driverReferral' -> pure $ PerformanceRes (Results (driverReferral'.referredCustomerCount) (driverReferral'.activatedCustomerCount))
+    Nothing -> throwError $ InvalidRequest "Driver doesn't have referralCode"
```

Comment:
use `fromMaybeM` for this case:

`driverReferral <- SQD.findById driverId >>= fromMaybeM (InvalidRequest "Driver doesn't have referralCode")`